rb_ws/src/buggy/buggy/serial/ros_to_bnyahaj.py

```diff
@@ -0,0 +1,255 @@
+            self.odom_publisher = self.create_publisher(
+                ROSOdom, self_name + "/nav/odom", 1
+            )
+        else:
+            self.odom_publisher = self.create_publisher(
+                ROSOdom, other_name + "/nav/odom", 1
+            )
+
+        # upper bound of steering update rate, make sure auton sends slower than 500 Hz or update / 2ms
+        self.steer_send_rate = self.create_rate(500)
+
+        # upper bound of reading data from Bnyahaj Serial, at 1ms
+        self.read_rate = self.create_rate(1000)
+
+        # ISSUE: https://github.com/CMU-Robotics-Club/RoboBuggy2/issues/90
+        self.rc_steering_angle_publisher = self.create_publisher(
+            Float64, self_name + "/buggy/debug/rc_steering_angle", 1
+        )
+        self.steering_angle_publisher = self.create_publisher(
+            Float64, self_name + "/buggy/debug/steering_angle", 1
+        )
+        self.battery_voltage_publisher = self.create_publisher(
+            Float64, self_name + "/buggy/debug/battery_voltage", 1
+        )
+        self.operator_ready_publisher = self.create_publisher(
+            Bool, self_name + "/buggy/debug/operator_ready", 1
+        )
+        self.steering_alarm_publisher = self.create_publisher(
+            Bool, self_name + "/buggy/debug/steering_alarm", 1
+        )
+        self.brake_status_publisher = self.create_publisher(
+            Bool, self_name + "/buggy/debug/brake_status", 1
+        )
+        self.use_auton_steer_publisher = self.create_publisher(
+            Bool, self_name + "/buggy/debug/use_auton_steer", 1
+        )
+        self.rc_uplink_qual_publisher = self.create_publisher(
+            UInt8, self_name + "/buggy/debug/rc_uplink_quality", 1
+        )
+        self.nand_fix_publisher = self.create_publisher(
+            UInt8, self_name + "/buggy/debug/nand_fix", 1
+        )
+
+    def set_alarm(self, msg):
+        """
+        alarm ros topic reader, locked so that only one of the setters runs at once
+        """
+        with self.lock:
+            self.get_logger().debug(f"Reading alarm of {msg.data}")
+            self.alarm = msg.data
+
+    def set_steering(self, msg):
+        """
+        Steering Angle Updater, updates the steering angle locally if updated on ros stopic
+        """
+        self.get_logger().debug(f"Read steeering angle of: {msg.data}")
+        # print("Steering angle: " + str(msg.data))
+        # print("SET STEERING: " + str(msg.data))
+        with self.lock:
+            self.steer_angle = msg.data
+            self.fresh_steer = True
+
+    def writer_thread(self):
+        """
+        Sends ROS Topics to bnayhaj serial, only sends a steering angle when we receive a fresh one
+        Will send steering and alarm node.
+        TODO: Does alarm node exist for NAND?
+        """
+        self.get_logger().info("Starting sending alarm and steering to teensy!")
+
+        while not self.is_shutdown():
```

Comment:
Here we have while not self.is_shutdown(), but in reader_tread, we have while rclpy.ok(), decide one of the 2 loops.